konami/kpi.md

```diff
@@ -0,0 +1,190 @@
+# Konami Product Identifier (KPI)
+
+a.k.a. "mcodes"
+
+For identification purpose, Konami labels their products, e.g. full cabinet assembly, hardware +
+software upgrade kit, software update, using a propriatary format. These "product codes" are often
+seen denoting specific versions/iterations of software or hardware.
+
+## Sources
+
+* pigstall
+* [bemaniwiki](http://bemaniwiki.com/)
+* Game/cabinet manuals
+
+## Product identifier format
+
+Format string: `ptggg:d:s:r:e`
+
+* `p`: product type
+  * `G`: game (?)
+* `t`: license type/eamuse participation
+  * `C`: serial, roundplug, ikey / software shipped with cabinet
+  * `E`: roundplug, ikey, etoken / upgrade kit or conversion kit
+  * `K`: offline capability (?) / offline upgrade kit
+  * `Q`: serial, ikey, etoken / rental unit (?)
+  * `U`: etoken
+* `ggg`: 3 digit gamecode, see [gamecodes and revisions section](#gamecodes-and-revisions)
+* `d`: dest (locale)
+  * `J`: Japan
+  * `U`: US
+  * `E`: Europe
+  * `K`: Korea
+  * `A`: Asia
+  * `Y`: Indonesia
+  * `C`: China
+* `s`: spec (specification), describes a specific cabinet type and combination of hardware, e.g.
+  PCB and IO boards
+  * `A`, `B`, `C`, ...
+* `r` rev (revision), software revision
+  * `A`, `B`, `C`, ...
+* `e`: ext (build date)
+  * Format: `yyyymmddnn`
+    * `yyyy` = year, e.g. `2020`
+    * `mm` = month, e.g. `11` for November
+    * `dd` = day of month, e.g. `03`
+    * `nn` = build number on the day, e.g. `00`, `01`, ...
+
+Note that older games are lacking the ext part/build date wherelse newer games with the ext part
+have their revisions pinned to `A`.
+
+Example: `GQLDJ:J:A:A:2011050900`
+
+* product type: `G`
+* license type: `Q`
+* [gamecode](#gamecodes-and-revisions): `LDJ` (Beatmania IIDX)
+  * project start year `L` = `2011`
+  * remainder: `DJ`
+* locale: `J` = Japan
+* cab specification: `A`
+* revision: `A`
+* build date: `2011/05/09`, `00` = first build of the day
+
+## Gamecodes and revisions
```

Comment:
I'd prefer sorting the gamecodes by gamecode, instead of categorizing per game series they may belong to.